rspirv/binary/decoder.rs

```diff
@@ -414,33 +409,29 @@ mod tests {
 
     #[test]
     fn test_decode_int64() {
-        let b = vec![0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef];
+        let b = [0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef];
         let mut d = Decoder::new(&b);
         assert_eq!(Ok(0xefcdab9078563412), d.int64());
-
-        let b = f32_to_bytes(-12.34);
-        let mut d = Decoder::new(&b);
-        assert_eq!(Ok(-12.34), d.float32());
     }
 
     #[test]
     fn test_decode_float32() {
-        let b = f32_to_bytes(42.42);
+        let b = [0x14, 0xAE, 0x29, 0x42];
```

Comment:
would be a little better if there was a conversion from `42.42` here instead of the hard-coded binary

Reply:
The problem with that is you'll just end up with the same code in the test as is in the code it's testing which seems a bit pointless, but I agree it isn't very nice. It could probably be improved if the decoder accepted words because at least then the input would be in a SPIR-V format rather than just seemingly random bytes.